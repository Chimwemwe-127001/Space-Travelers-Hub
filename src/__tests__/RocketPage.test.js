import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { fireEvent, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import { Provider } from 'react-redux';
import store from '../redux/configureStore'
import RocketPage from '../component/pages/rocketpage/RocketPage';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('testing calculator and events', () => {
  it('testing for events on calculator', () => {
    act(() => {
      render(<Provider store={store}> <RocketPage /> </Provider>, container);
    });

    expect(pretty(container.innerHTML)).toMatchSnapshot();

    fireEvent.click(screen.getAllByText('Reserve Rocket'));

    let displayValue;

    await waitFor(() => {
      const displayElement = container.innerHTML.querySelectorAll('.reserve-btn');
    });

    expect(displayValue).toEqual('5');
  })
})