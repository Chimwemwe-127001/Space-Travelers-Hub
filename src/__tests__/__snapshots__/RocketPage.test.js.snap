// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing calculator and events testing for events on calculator 1`] = `" <div class=\\"rocket-page d-flex a-center j-center f-col\\"></div>"`;
